models/pyannote.py

```diff
@@ -0,0 +1,58 @@
+# pip install pyannote.metrics
+# pip install pyannote.pipeline
+# pip install pescador
+# pip install optuna
+# pip install filelock
+# pip install git+https://github.com/pyannote/pyannote-audio
+# pip install sortedcontainers simplejson typing_extensions pyannote.database pyannote.metrics
+# pip install pyannote.core --no-dependencies
+
+import shapes
+import torch
+import numpy as np
+
+
+class PyannoteDiarizationModel:
+	def __init__(self, vad, vad_sensitivity: float, device: str, sample_rate: int):
+		assert sample_rate == 16_000, 'Only 16 kHz sample rate supported'
+		self.vad = vad
+		self.vad_sensitivity = vad_sensitivity
+		self.device = device
+		self.sample_rate = sample_rate
+		self.model = torch.hub.load('pyannote/pyannote-audio', 'dia', device = torch.device(device), batch_size = 1)
+
+	def get_silence_mask(self, signal: shapes.BT):
+		smax = np.iinfo(np.int16).max
+		f2s = lambda signal, max=np.float32(smax): torch.mul(signal, max).to(torch.int16)
+		s2f = lambda signal, max=np.float32(smax): torch.div(signal, max, dtype = torch.float32)
+
+		if signal.dtype == torch.float32 and self.vad.required_type == 'int16':
```

Comment:
What is semantics of `dtype` and `required_type`? Can we handle audio conversion outside of the model?

Reply:
`required_type` -> `input_dtype`

Reply:
Move `input_dtype` property to model